### PR TITLE
Add task logging

### PR DESCRIPTION
**Pull request by linonymous**
Signed-off-by: linonymous <iamlinonymous@gmail.com>

### CODE REVIEW

**Review by lukaszgryglicki — APPROVED** [review]
/lgtm - but I would think about wrapping that logger calls in `shared.Printf` function - that way we will have all current console logging going to that new logger out of the box. We could also add a flag to enable/disable this - so, depending on that flag `shared.printf` would do an extra logger call or not. What do you think? @linonymous @fayazg @1010sachin ?

**Comment by linonymous** [comment]
> /lgtm - but I would think about wrapping that logger calls in `shared.Printf` function - that way we will have all current console logging going to that new logger out of the box. We could also add a flag to enable/disable this - so, depending on that flag `shared.printf` would do an extra logger call or not. What do you think? @linonymous @fayazg @1010sachin ?

yep this can be done. But, this specific logging is done for scheduling purpose so that the scheduler can understand what jobs has finished & what jobs are still in_progress.

**Comment by lukaszgryglicki** [comment]
Sure, I've approved, but would be great to wrap around `shared.Printf` too - and make it configurable via a flag.

**Comment by lukaszgryglicki** [comment]
Can we get this merged?

File: cmd/git/git.go [hunk 1]
```diff
@@ -553,17 +561,46 @@ type DSGit struct {
...
+}
+
+func (j *DSGit) WriteLog(ctx *shared.Ctx, status, message string) {
+	_ = j.Logger.Write(&logger.Log{
+		Connector:     GitDataSource,
+		Configuration: []map[string]string{{"REPO_URL": j.URL, "ES_URL": ctx.ESURL}},
+		Status:        status,
+		CreatedAt:     time.Now(),
+		UpdatedAt:     time.Now(),
+		ProjectSlug:   ctx.Project,
```
Comment (khalifapro) [inline cmd/git/git.go:599]: project slug should be in configuration slice, you will need to update shared version

**Comment by lukaszgryglicki** [comment]
Please ping me once merged `git` and `github` connectors will need changes to address @khalifapro 's comments, I don't want to miss this.

File: cmd/git/git.go (hunk 1 above)
Comment (linonymous) [reply cmd/git/git.go]: done

File: cmd/git/git.go [hunk 2]
```diff
@@ -77,6 +79,12 @@ const (
 	CommitCreated = "commit.created"
 	// CommitUpdated commit updated event
 	CommitUpdated = "commit.updated"
+	// InProgress status
+	InProgress = "in_progress"
```
[the remark this answers is not in the document]
Comment (linonymous) [reply cmd/git/git.go:84]: done.

**Comment by linonymous** [comment]
Pull request to provide the Elastic log URLs & other env variables when the container spawns: https://github.com/LF-Engineering/lfx-insights/pull/305 cc: @lukaszgryglicki

**Pull request merged**